### PR TITLE
Design

### PR DESCRIPTION
**Pull request by jmy-kellogg**
most things are fixed. The add question button doesn't appear when there are no questions on top layer. Please don't merge until someone test out that it's working okay.

### CODE REVIEW

**Comment by DougVonMoser** [comment]
SWEET!! this looks really cool and intuitive

File: businessServer/client/components/Product.js [hunk 1]
```diff
@@ -27,11 +27,11 @@ const Product = React.createClass({
...
     // var price = +this.refs.price.value;
     // var description = this.refs.description.value;
-// <<<<<<< HEAD
-// =======
-    var price = null;
-    var description = null;
-// >>>>>>> master
+
+//     var price = null;
+//     var description = null;
```
Comment (wjrhee) [inline businessServer/client/components/Product.js]: don't comment out.  bring it back.
var price = null
var description = null

it was left in there just in case we wanted to add in the price and description feature.  there might be other things that are dependent on those variables.

File: businessServer/client/components/Product.js [hunk 2]
```diff
@@ -74,14 +74,9 @@ const Product = React.createClass({
...
       return (
-//<<<<<<< HEAD
-        //<div className={divClassName} key={i} onClick={this.selectProduct.bind(this, product)}>
-          // <h3>{product.name}</h3>
-// {/*=======*/}
-
-        <div className="product-div" key={i} onClick={this.selectProduct.bind(this, product)}>
-           {product.name}
-{/*>>>>>>> master*/}
+
```
Comment (wjrhee) [inline businessServer/client/components/Product.js:68]: remove the '<<<<<< HEAD' and '>>>>> master' comments.  and lines 77-79

File: businessServer/client/components/SingleForm.js [hunk 3]
```diff
@@ -96,24 +98,23 @@ const SingleForm = React.createClass({
...
-    // }
-    // else{
-    //   nextTreePointer = null;
-    // }
+    if(this.props.topLevelNodeIndex < this.props.topLevelNodes[this.props.prodSelected].length-1){
+      nextTreePointer = this.props.topLevelNodes[this.props.prodSelected][this.props.data.topLevelNodeIndex+1];
+    }
+    else{
+      nextTreePointer = null;
+    }
```
Comment (wjrhee) [inline businessServer/client/components/SingleForm.js]: leave this code commented out.  101-106.
the nextTreePointer is assigned in 109-114.

File: businessServer/client/components/topLayer.js [hunk 4]
```diff
@@ -4,36 +4,19 @@ import InlineEdit from './InlineEdit';
...
-    console.log(this.props.prodSelected);
-    console.log(this.props.topLevelNodes);
-    var newTopLevelIndex = 0;
-    if(this.props.topLevelNodes[this.props.prodSelected]){
-      newTopLevelIndex = this.props.topLevelNodes[this.props.prodSelected].length;
-    }
-    // var newTopLevelIndex = this.props.topLevelNodes[this.props.prodSelected].length;
-
-// >>>>>>> master
     e.preventDefault();
```
Comment (wjrhee) [inline businessServer/client/components/topLayer.js:27]: please bring back lines 14-19 here.

File: businessServer/client/components/topLayer.js [hunk 5]
```diff
@@ -4,36 +4,19 @@ import InlineEdit from './InlineEdit';
...
     axios.post('/api/nodes', {
       question: "default question",
       productId: this.props.prodSelected,
       topLevel: true,
-      layer: 1,
-      topLevelNodeIndex: newTopLevelIndex,
-      leafNode: true
+      layer: 1
     })
```
Comment (wjrhee) [inline businessServer/client/components/topLayer.js:35]: bring back lines 28-30.
posting layer: 1
topLevelNodeIndex
and LeafNode

File: businessServer/client/components/topLayer.js [hunk 6]
```diff
@@ -4,36 +4,19 @@ import InlineEdit from './InlineEdit';
...
     .then(node => node.data)
     .then(node => {
-      console.log(newTopLevelIndex)
-
-      this.props.addNewNode(node.productId, node.id, 1, true, node.productId, newTopLevelIndex, true);
+    // console.log(this.props);
+      this.props.addNewNode(node.productId, node.id, 1, true, node.productId);
       return node;
```
Comment (wjrhee) [inline businessServer/client/components/topLayer.js:41]: bring back

File: businessServer/client/components/topLayer.js [hunk 7]
```diff
@@ -56,27 +39,8 @@ const TopLayer = React.createClass({
...
-      if(node.question){
-        q = node.question;
-      }
-      else{
-        q = "I'm a question? Fill me out.";
-      }
-     return (
-      <div key={i} id={`nodeContainer${i}`} onClick={this.scrollTo.bind(this, i)}>
-{/*=======
 render: function(){
```
Comment (wjrhee) [inline businessServer/client/components/topLayer.js:56]: why was this removed?

File: businessServer/client/components/topLayer.js [hunk 8]
```diff
@@ -94,23 +58,17 @@ render: function(){
...
-{/*<<<<<<< HEAD
-    {nodesDiv}
-    <div className='addtoplayernode' onClick={this.addTopLayerNode}><span className="glyphicon glyphicon-plus"></span></div>
-
-{/*=======*/}
-    {this.props.prodSelected !== undefined ? <div className='metal addtoplayernode' onClick={this.addTopLayerNode}> Add New Question</div> : null}
       {nodesDiv}
-{/*>>>>>>> master*/}
+      {nodesDiv.length > 0 ? <div className='addtoplayernode' onClick={this.addTopLayerNode}> <span className="glyphicon glyphicon-plus"></span></div> : null}
    </div>
```
Comment (wjrhee) [inline businessServer/client/components/topLayer.js:83]: i know this logic is broken.  this is fine.  i'll fix it later

**Comment by jmy-kellogg** [comment]
I think everything is resolved. Hopefully there are no bugs that I missed.
